capa/rules/__init__.py

```diff
@@ -215,20 +251,31 @@ def __repr__(self):
         return str(self)
 
 
-def ensure_feature_valid_for_scope(scope: str, feature: Union[Feature, Statement]):
+def ensure_feature_valid_for_scope(scope: Union[str, Flavor], feature: Union[Feature, Statement]):
```

Comment:
throughout this PR i would remove all instances of the `Union[str, Flavor]` and use just `Flavor` (or `Scopes`, as proposed). 

using a Union means that we have to check for the two options in many different functions and be sure to handle every case everywhere. instead, i think we should commit to the new API and remove old behaviors completely.

Reply:
unrelated: this function checks the features for a single scope, so i'd recommend that it receive `scope: Scope` as an argument, not `scope: Flavor/Scopes`. if there's logic that needs to be introduced for `Scopes`, add a new function.

try to keep it clear:
  - `Scope` a single scope, like "function"
  - `Scopes` a mapping of flavors to scopes, like `{static: function, dynamic: thread}`

Reply:
furthermore, we may want to update the rule syntax a bit more: rename the `meta.scope` field to `meta.scopes` to align it with our type structure in this PR. because we expect to have more than one entry in that part of the metadata, it makes sense for the term to be plural.

i think that then our code will be pretty consistent throughout, at least around using scope/scopes.